typescript/src/merchi.ts

```diff
@@ -20,9 +24,17 @@ export class Merchi {
   public id: string = generateUUID();
 
   public Category: typeof Category
+  public Domain: typeof Domain;
   public Product: typeof Product;
 
-  constructor() {
+  public sessionToken?: string;
+
+  constructor(sessionToken?: string) {
+    if (sessionToken) {
```

Comment:
I was struggled when look at it, I think you might be struggled to make this decision as well.
I think it is fine for now

Reply:
looking up cookies using document.cookie is a bit of a hack, in particular, i don't know how well that's going to work on node or a another non-browser environment.

but, line 33 that you pointed out, i thought it was quite clear. straight forward check for whether or not the argument was provided.... what part don't you like?